### ISSUE TITLE
Running it in container scheduler

### ISSUE THREAD

**Issue by rokka-n** [erlio/docker-vernemq#16]
Hi,

I'm trying to run it in a scheduler (nomad), but... it's not running.

First I'm not sure about ranges for port allocation. Is it possible to run it with specific port mapping - which looks like the case from the readme?

Then I wonder if it can do retry for discovery? I'd like to have this work: 
`DOCKER_VERNEMQ_DISCOVERY_NODE="mqtt-${env}.service.consul"`

I've also tried to run it with net=host, but looks like it requires rewriting all the startup script. Not sure, maybe it' scheduler.

```
2017-04-25 23:51:21.167 [info] <0.31.0> Application vmq_server started on node 'VerneMQ@172.17.0.2'
2017-04-25 23:51:21.200 [info] <0.3.0>@plumtree_peer_service:attempt_join:50 Sent join request to: 'VerneMQ@mqtt-dev.service.consul'
2017-04-25 23:51:21.200 [debug] <0.424.0> Supervisor inet_gethost_native_sup started undefined at pid <0.425.0>
2017-04-25 23:51:21.200 [debug] <0.59.0> Supervisor kernel_safe_sup started inet_gethost_native:start_link() at pid <0.424.0>
2017-04-25 23:51:21.223 [info] <0.3.0>@plumtree_peer_service:attempt_join:53 Unable to connect to 'VerneMQ@mqtt-dev.service.consul'
2017-04-25 23:51:24.280 [debug] <0.434.0>@vmq_ranch:teardown:123 session normally stopped
```

**Comment by rokka-n** [comment]
I also can run a "special" node just for discovery, but I guess that would be a waste of resources.

**Comment by dergraf** [comment]
We've had a discussion regarding node discovery mechanism over here. #11 
TL;DR; we currently don't have a good way to deal with automatic cluster discovery.
